backend/packages/Upgrade/src/api/repositories/FeatureFlagRepository.ts

```diff
@@ -111,4 +112,19 @@ export class FeatureFlagRepository extends Repository<FeatureFlag> {
 
     return result;
   }
+
+  public async validateUniqueKey(flagDTO: FeatureFlagValidation) {
+    const result = await this.createQueryBuilder('feature_flag')
+      .where('feature_flag.key = :key', { key: flagDTO.key })
+      .andWhere('feature_flag.context = :context', { context: flagDTO.context })
+      .getOne();
+
+    if (result) {
```

Comment:
This part of the code should be in the service as it is not releated to data fetching

Reply:
code moved